compiler/rustc_target/src/spec/mod.rs

```diff
@@ -3067,9 +3067,9 @@ impl Target {
         }
 
         check_eq!(
-            self.is_like_osx,
+            self.is_like_darwin,
             self.vendor == "apple",
-            "`is_like_osx` must be set if and only if `vendor` is `apple`"
+            "`is_like_darwin` must be set if and only if `vendor` is `apple`"
```

Comment:
it's there a case where `is_like_darwin` is false but vendor is still Apple? out could we turn those into a function which compares the vendor?

Reply:
No such cases, no.

But it seems to me like all the other `is_like_*` fields should also be functions, then?

Reply:
I guess?..